apps/annotations/views/tag_views.py

```diff
@@ -61,17 +61,26 @@ class EditTag(UpdateView, PermissionRequiredMixin):
     }
 
     def get_queryset(self):
-        return Tag.objects.filter(team=self.request.team)
+        return Tag.objects.filter(team=self.request.team, is_system_tag=False)
 
     def get_success_url(self):
         return reverse("annotations:tag_home", args=[self.request.team.slug])
 
+    def get(self, request, *args, **kwargs):
+        pk = kwargs.get("pk")
+        queryset = self.get_queryset()
+        if not queryset.filter(pk=pk).exists():
+            return HttpResponseRedirect(reverse("annotations:tag_home", args=[self.request.team.slug]))
+        return super().get(request, *args, **kwargs)
+
 
 class DeleteTag(LoginAndTeamRequiredMixin, View, PermissionRequiredMixin):
     permission_required = "annotations.delete_tag"
 
     def delete(self, request, team_slug: str, pk: int):
         tag = get_object_or_404(Tag, id=pk, team=request.team)
+        if tag.is_system_tag:
+            return HttpResponseForbidden("System tags cannot be deleted.")
```

Comment:
How is this show in the UI?

Reply:
with the deletion option taken away this ideally would never show in the UI but rather serve only as a backup, but can remove

Reply:
Fine to leave it as a safeguard